### ISSUE TITLE
feature request: resample track/seqblock

### ISSUE THREAD

**Issue by autumncheney** [kmatheussen/radium#1414]
it would be cool if you could directly resample tracks and seqblocks. if you don't know what resampling is, it's basically recording something within a daw and then re-importing it back into the daw. in radium this would probably either be a sample player instrument or a sample in a seqtrack

this would be a lot easier than manually creating a sample player and then routing anything you'd want resampled into it via the modular mixer. i imagine the created sample could automatically get its own track in the seqblock, or a timeline sample if it's the entire seqblock

**Comment by kmatheussen** [comment]
I'm not sure if I understand what you mean. If you change the "Speed" values in the seqblocks (the red knobs, below "Fade" and "--|", and above "Stretch"), you change how much a sample is resampled.

**Comment by autumncheney** [comment]
[this is what i mean by "resampling"](https://youtu.be/eU7Qmm6FCEI?feature=shared&t=37) (it's in ableton live)

he's taking the audio output of the daw and recording it into the track. this is the basic principle of resampling

**Comment by kmatheussen** [comment]
Ah, that's completely different than what is traditionally meant by "resampling". Very confusing. :-)

Anyway, you can record anything from the mixer into a seqtrack. Just rightclick the "R" button in the seqtrack header, and select "Input connections to the instrument" as "Source". Then patch what you want to record in the mixer and so forth. I think that's what you meant?

**Comment by kmatheussen** [comment]
So, to do what that guy did in the video:
1. Disconnect the output of the seqtrack and the "Main Bus" (to prevent recursive connections).
2. Connect the output of the "Main bus" and the input of the seqtrack.
3. Solo/mute instruments you want to hear/not hear.
4. Left-click "R" on the seqtrack
5. Play

**Comment by autumncheney** [comment]
thank you! though it would be easier if there was a way to automatically change the connections as required for this process. i suppose this is what i was originally requesting

**Comment by autumncheney** [comment]
incidentally, is there a way to only play the song/seqblock/range once without looping? when i tested this method, the song continued to play even after it played through

**Comment by kmatheussen** [comment]
Not sure if I know what you mean. If you have enabled looping in the sequencer, it will always loop. If you haven't enabled looping, it should never loop.

**Comment by kmatheussen** [comment]
Maybe you're using looping instead of "Punch in/out"?

**Comment by kmatheussen** [comment]
> thank you! though it would be easier if there was a way to automatically change the connections as required for this process. i suppose this is what i was originally requesting

You mean disconnecting seqtrack->main bus and connect main bus->seqtrack in one step?

**Comment by autumncheney** [comment]
> > thank you! though it would be easier if there was a way to automatically change the connections as required for this process. i suppose this is what i was originally requesting
> 
> You mean disconnecting seqtrack->main bus and connect main bus->seqtrack in one step?

yes. and changing it back recording is finished

**Comment by autumncheney** [comment]
> Maybe you're using looping instead of "Punch in/out"?

punch in/out will work for now

**Comment by kmatheussen** [comment]
In the video, though, he didn't connect main bus, I think he connected what was sent to "System out".

**Comment by kmatheussen** [comment]
> > > thank you! though it would be easier if there was a way to automatically change the connections as required for this process. i suppose this is what i was originally requesting
> > 
> > 
> > You mean disconnecting seqtrack->main bus and connect main bus->seqtrack in one step?
> 
> yes. and changing it back recording is finished

Right. I guess this could be a common operation so it would make sense to make it quicker.

**Comment by kmatheussen** [comment]
> In the video, though, he didn't connect main bus, I think he connected what was sent to "System out".

Or probably no, then all the global effects would have been applied twice.

**Comment by autumncheney** [comment]
> > In the video, though, he didn't connect main bus, I think he connected what was sent to "System out".
> 
> Or probably no, then all the global effects would have been applied twice.

actually it would be nice if you could change from what point in the audio graph the resampling is done from some kind of menu, so effects need not be applied twice like you said

**Comment by kmatheussen** [comment]
Wouldn't it be better to use "Export soundfile(s)"? 
There isn't an option to record the punch in/out range, but if there were, then it would be much quicker.

(Also, please don't use the word "resample" for this. Resampling means changing sample rate.)

**Comment by kmatheussen** [comment]
> Resampling means changing sample rate.)

Well, not really, but that's what most people immediately associate with the word I would think.

**Comment by kmatheussen** [comment]
In Protools, this is called "Bounce". I think we should use that word.

**Comment by autumncheney** [comment]
ardour uses "bounce" as well, so i think that terminology is a good idea

anyways, the export soundfile dialog works as well! i didn't realize it could export a block or range only. though maybe the bounce menu could allow automatically exporting and then loading the sample into the seqtrack/sample player? as well as any needed connection changes involved?
